src/neptune_scale/core/components/aggregating_queue.py

```diff
@@ -118,10 +130,12 @@ def get_nowait(self) -> BatchedOperations:
                 batch_timestamp = element.timestamp
 
                 self.commit()
-        else:
-            logger.debug("Batch closed due to queue being empty")
+
+            t1 = time.monotonic()
```

Comment:
You can just store `t0` and check the remaining time in the condition of `while` directly